### PR TITLE
[PAR-3817] Add New Extension Attributes for Shipping Protection

### PR DESCRIPTION
**Pull request by jomarsantos**
- Switches extension attribute type from array to ShippingProtectionTotalInterface for Orders

### CODE REVIEW

**Review by jay-faust-extend — APPROVED** [review]

**Review by scott-taylor-extend — CHANGES_REQUESTED** [review]
Isn't there more files that need to be updated as well to switch from an array to an object?

File: Plugin/Model/OrderRepositoryPlugin.php [hunk 1]
```diff
@@ -51,15 +51,7 @@ public function afterGet(\Magento\Sales\Model\OrderRepository $subject, $result,
...
-        $extensionAttributes->setShippingProtection(
-            [
-                'base' => $shippingProtectionTotal->getShippingProtectionBasePrice(),
-                'base_currency' => $shippingProtectionTotal->getShippingProtectionBaseCurrency(),
-                'price' => $shippingProtectionTotal->getShippingProtectionPrice(),
-                'currency' => $shippingProtectionTotal->getShippingProtectionCurrency(),
-                'sp_quote_id' => $shippingProtectionTotal->getSpQuoteId()
-            ]
-        );
+        $extensionAttributes->setShippingProtection($shippingProtectionTotal);
```
Comment (scott-taylor-extend) [inline Plugin/Model/OrderRepositoryPlugin.php]: This would break line 82 on this same file as it's looking for the array values still not an object.  That line would need to be updated.  This is just a start.  There is a lot more changes needed for this to work

File: Observer/SaveOrderBeforeSalesModelQuote.php [hunk 2]
```diff
@@ -65,9 +65,6 @@ public function execute(Observer $observer)
             if ($extensionAttributes === null) {
                 $extensionAttributes = $this->orderExtensionFactory->create();
             }
-            if ($extensionAttributes->getShippingProtection() === null) {
```
Comment (scott-taylor-extend) [inline Observer/SaveOrderBeforeSalesModelQuote.php]: Why did you remove this?

**Review by scott-taylor-extend — CHANGES_REQUESTED** [review]

File: Observer/SaveOrderBeforeSalesModelQuote.php (hunk 2 above)
Comment (jomarsantos) [reply Observer/SaveOrderBeforeSalesModelQuote.php]: Since it is an class now, `null` seems more fitting since it isn't an associative array anymore

**Comment by jomarsantos** [comment]
@scott-taylor-extend I'm thinking maybe we do need to do all of the changes at once due to areas like this:
https://github.com/helloextend/magento-extend-integration/blob/adba5cad41ec9913c27d85cea2348a2df1133331/Block/Adminhtml/Sales/Order/Creditmemo/Totals.php#L78-L99

**Comment by jomarsantos** [comment]
@scott-taylor-extend decided to go a simpler route - let me know what you think

**Comment by scott-taylor-extend** [comment]
> 

@jomarsantos I'd rather stick with the route we were on, with the extension_attribute model.  I can take it over today if you want.
